### PR TITLE
DDT-1202: vessel position requirement

### PR DESCRIPTION
**Pull request by Abdiiir**

### CODE REVIEW

File: src/app/view/timestamp-editor/timestamp-editor.component.ts [hunk 1]
```diff
@@ -140,8 +140,16 @@ export class TimestampEditorComponent implements OnInit {
...
+    const timestampTypeSelected = this.timestampTypeSelected?.value; 
+    if (timestampTypeSelected?.vesselPositionRequirement === EventLocationRequirement.REQUIRED){
+      validators = [Validators.required]; 
+    }
+      this.timestampFormGroup.controls.vesselPositionLatitude.setValidators(validators);
+      this.timestampFormGroup.controls.vesselPositionLongitude.setValidators(validators);
+      this.timestampFormGroup.controls.vesselPositionLatitude.updateValueAndValidity();
+      this.timestampFormGroup.controls.vesselPositionLongitude.updateValueAndValidity();
+      return timestampTypeSelected?.vesselPositionRequirement !== EventLocationRequirement.EXCLUDED;
   }
```
Comment (nt-as) [inline src/app/view/timestamp-editor/timestamp-editor.component.ts:153]: Weird indentation, please realign here.

File: src/app/view/transport-call-creator/transport-call-creator.component.ts [hunk 2]
```diff
@@ -205,9 +205,17 @@ export class TransportCallCreatorComponent implements OnInit {
...
+    if (timestampTypeSelected?.vesselPositionRequirement === EventLocationRequirement.REQUIRED 
+      && this.timestampChecking){
+      validators = [Validators.required]; 
+    }
+      this.transportCallFormGroup.controls.vesselPositionLatitude.setValidators(validators);
+      this.transportCallFormGroup.controls.vesselPositionLongitude.setValidators(validators);
+      this.transportCallFormGroup.controls.vesselPositionLatitude.updateValueAndValidity();
+      this.transportCallFormGroup.controls.vesselPositionLongitude.updateValueAndValidity();
+      return timestampTypeSelected?.vesselPositionRequirement !== EventLocationRequirement.EXCLUDED;
   }
```
Comment (nt-as) [inline src/app/view/transport-call-creator/transport-call-creator.component.ts:219]: Ditto, weird indentation.

Comment (Abdiiir) [reply src/app/view/transport-call-creator/transport-call-creator.component.ts:219]: Should probably have this done automatically heheh

File: src/app/view/timestamp-editor/timestamp-editor.component.ts [hunk 3]
```diff
@@ -144,8 +144,16 @@ export class TimestampEditorComponent implements OnInit {
...
+    const timestampTypeSelected = this.timestampTypeSelected?.value;
+    if (timestampTypeSelected?.vesselPositionRequirement === EventLocationRequirement.REQUIRED) {
+      validators = [Validators.required];
+    }
+    this.timestampFormGroup.controls.vesselPositionLatitude.setValidators(validators);
+    this.timestampFormGroup.controls.vesselPositionLongitude.setValidators(validators);
+    this.timestampFormGroup.controls.vesselPositionLatitude.updateValueAndValidity();
+    this.timestampFormGroup.controls.vesselPositionLongitude.updateValueAndValidity();
+    return timestampTypeSelected?.vesselPositionRequirement !== EventLocationRequirement.EXCLUDED;
   }
```
Comment (nt-as) [inline src/app/view/timestamp-editor/timestamp-editor.component.ts:157]: Getter should not have side-effects.

File: src/app/view/transport-call-creator/transport-call-creator.component.ts [hunk 4]
```diff
@@ -205,9 +205,17 @@ export class TransportCallCreatorComponent implements OnInit {
...
+    if (timestampTypeSelected?.vesselPositionRequirement === EventLocationRequirement.REQUIRED
+      && this.timestampChecking) {
+      validators = [Validators.required];
+    }
+    this.transportCallFormGroup.controls.vesselPositionLatitude.setValidators(validators);
+    this.transportCallFormGroup.controls.vesselPositionLongitude.setValidators(validators);
+    this.transportCallFormGroup.controls.vesselPositionLatitude.updateValueAndValidity();
+    this.transportCallFormGroup.controls.vesselPositionLongitude.updateValueAndValidity();
+    return timestampTypeSelected?.vesselPositionRequirement !== EventLocationRequirement.EXCLUDED;
   }
```
Comment (nt-as) [inline src/app/view/transport-call-creator/transport-call-creator.component.ts:219]: Getter should not have side-effects.

File: src/app/view/timestamp-editor/timestamp-editor.component.ts (hunk 3 above)
Comment (Abdiiir) [reply src/app/view/timestamp-editor/timestamp-editor.component.ts:157]: This is how we have been defining our "show" methods on the component, for example the method `showTerminalOption()`

Comment (nt-as) [reply src/app/view/timestamp-editor/timestamp-editor.component.ts:157]: "Two wrongs do not make a right".

**Review by nt-as — APPROVED** [review]

**Review by nt-as — DISMISSED** [review]

**Review by nt-as — APPROVED** [review]

**Pull request merged**